javaslang/src/main/java/javaslang/collection/Traversable.java

```diff
@@ -981,6 +986,21 @@ default T reduceRight(BiFunction<? super T, ? super T, ? extends T> op) {
         return isEmpty() ? Option.none() : Option.some(reduceRight(op));
     }
 
+    @Override
+    default Spliterator<T> spliterator() {
+        int characteristics = Spliterator.IMMUTABLE;
+        if (isDistinct()) {
+            characteristics |= Spliterator.DISTINCT;
+        }
+        if (isOrdered()) {
+            characteristics |= (Spliterator.ORDERED | Spliterator.SORTED);
+        }
```

Comment:
I think we need an additional check for sequential collection (= insertion-order):

```java
if (isSequential()) {
    characteristics |= Spliterator.ORDERED;
}
```

---

**Specs:**

```java
interface Traversable<T> {
    /**
     * Checks if the elements of this Traversable appear in encounter order.
     *
     * @return true, if the insertion order of elements is preserved, false otherwise.
     */
    default boolean isSequential() {
        return false;
    }
}
```

Please insert the new method right below Traversable isOrdered().

Please add also a link to this method to the javadoc of the Traversable class:

```java
...
 * Tests:
 *
 * <ul>
 * <li>{@link #existsUnique(Predicate)}</li>
 * <li>{@link #hasDefiniteSize()}</li>
 * <li>{@link #isDistinct()}</li>
 * <li>{@link #isOrdered()}</li>
 * <li>{@link #isSequential()}</li>
 * <li>{@link #isTraversableAgain()}</li>
 * </ul>
...
```

The following collection types need to override `isSequential()` by returning `true`:

* `Seq`
* `LinkedHashSet`
* `LinkedHashMap`
* `LinkedHashMultimap`
* `Tree`
* `Iterator`